src/groupeddataframe/groupeddataframe.jl

```diff
@@ -303,7 +303,7 @@ Base.getindex(gd::GroupedDataFrame, idx...) =
     getindex(gd, Base.to_indices(gd, idx)...)
 
 # The allowed key types for dictionary-like indexing
-const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple}
+const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple, Dict}
```

Comment:
Defining it in this way should give a bit better error messages
```suggestion
const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple, Dict{Symbol}, Dict{<:AbstractString}}
```